File: app/src/main/java/com/example/komuroshun/androidtabui/HomeActivity.java
package com.example.komuroshun.androidtabui;

import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.FragmentTabHost;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Spinner;
import android.widget.TabHost;
import android.widget.TabWidget;

/**
 * Activities manage fragments.
 * @author Shun Komuro
 * @version 1.0
 */
public class HomeActivity extends AppCompatActivity implements FragmentTabHost.OnTabChangeListener,
        PageFragment.OnFragmentInteractionListener,
        TabContainerFragment.OnFragmentInteractionListener {

    private static final String TAG = HomeActivity.class.getName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);

        // Toolbar Settings
        Toolbar toolbar = (Toolbar) findViewById(R.id.tool_bar);
        toolbar.setTitle("");
        setSupportActionBar(toolbar);
        Spinner spinner = (Spinner) toolbar.findViewById(R.id.tool_bar_spinner);
//        spinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
//            @Override
//            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
//                String selectedName = String.valueOf(parent.getSelectedItem());
//                Bundle selectedNameBundle = new Bundle();
//                selectedNameBundle.putString("name", selectedName);
//                FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
//                Fragment tabContainerFragment = new TabContainerFragment();
//                tabContainerFragment.setArguments(selectedNameBundle);
//                fragmentTransaction.replace(R.id.container, tabContainerFragment);
//                fragmentTransaction.addToBackStack(null); // 戻るボタンでreplace前に戻る
//                fragmentTransaction.commit();
//            }
//            @Override
//            public void onNothingSelected(AdapterView<?> parent) {
//            }
//        });
        spinner.setAdapter(ArrayAdapter.createFromResource(this, R.array.fragment_names,
                android.R.layout.simple_spinner_dropdown_item));

        FragmentTabHost tabHost = findViewById(android.R.id.tabhost);
        tabHost.setup(this, getSupportFragmentManager(), R.id.container);

        // 画面下部タブメニュー
        TabHost.TabSpec tabHome, tabNotice, tabPoint, tabHistory, tabOthers;
        // タブ生成1
        View tabViewHome = new CustomBottomTabContentView(this, "ホーム", R.drawable.selector_bottom_tab_home);
        tabHome = tabHost.newTabSpec("ホーム");
        tabHome.setIndicator(tabViewHome);
        // クラス調査使い方
        Bundle bundle1 = new Bundle();
        bundle1.putString("name", "ホーム");
        tabHost.addTab(tabHome, PageFragment.class, bundle1);
        // タブ生成2
        View tabViewNotice = new CustomBottomTabContentView(this, "お知らせ", R.drawable.selector_bottom_tab_notice);
        tabNotice = tabHost.newTabSpec("お知らせ");
        tabNotice.setIndicator(tabViewNotice);
        Bundle bundle2 = new Bundle();
        bundle2.putString("name", "お知らせ");
        tabHost.addTab(tabNotice, PageFragment.class, bundle2);
        // タブ生成3
        View tabViewPoint = new CustomBottomTabContentView(this, "ポイント", R.drawable.selector_bottom_tab_point);
        tabPoint = tabHost.newTabSpec("ポイント");
        tabPoint.setIndicator(tabViewPoint);
        Bundle bundle3 = new Bundle();
        bundle3.putString("name", "ポイント");
        tabHost.addTab(tabPoint, PageFragment.class, bundle3);

        View tabViewHistory = new CustomBottomTabContentView(this, "履歴", R.drawable.selector_bottom_tab_history);
        tabHistory = tabHost.newTabSpec("履歴");
        tabHistory.setIndicator(tabViewHistory);
        Bundle bundle4 = new Bundle();
        bundle4.putString("name", "履歴");
        tabHost.addTab(tabHistory, TabContainerFragment.class, bundle4);

        View tabViewOthers = new CustomBottomTabContentView(this, "6Pack", R.drawable.selector_bottom_tab_others);
        tabOthers = tabHost.newTabSpec("6Pack");
        tabOthers.setIndicator(tabViewOthers);
        Bundle bundle5 = new Bundle();
        bundle5.putString("name", "6Pack");
        tabHost.addTab(tabOthers, TabContainerFragment.class, bundle5);

        // リスナー登録
        tabHost.setOnTabChangedListener(this);
    }

    @Override
    public void onTabChanged(String tabId) {
    }

    /**
     * Switch Fragments only.
     * @param uri Uri
     * @author Shun Komuro
     */
    @Override
    public void onFragmentInteraction(Uri uri) {
    }
}

File: app/src/main/java/com/example/komuroshun/androidtabui/HomeActivity.kt
package com.example.komuroshun.androidtabui

import android.net.Uri
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.FrameLayout
import kotlinx.android.synthetic.main.activity_home.*
import kotlinx.android.synthetic.main.widget_toolbar.*

/**
 * Activities manage fragments.
 * @author Shun Komuro
 * @version 1.0
 */
class HomeActivity : AppCompatActivity(), PageFragment.OnFragmentInteractionListener {

    companion object {
        const private val KEY_CURRENT_MENU_ID = "key_current_menu_id"
    }

    private var currentMenuId = 0
    private var currentTabId = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)
        toolbar.inflateMenu(R.menu.menu_toolbar)

        if (savedInstanceState == null) {
            currentMenuId = R.id.nav_home
        } else {
            currentMenuId = savedInstanceState.getInt(KEY_CURRENT_MENU_ID)
        }
        switchToSelectedTab(currentMenuId)

        BottomNavigationViewHelper.disableShiftMode(navigationBottom);
        navigationBottom.setOnNavigationItemSelectedListener { item ->
            switchToSelectedTab(item.itemId)
            return@setOnNavigationItemSelectedListener true
        }
    }

    override fun onSaveInstanceState(outState: Bundle?) {
        super.onSaveInstanceState(outState)
        outState!!.putInt(KEY_CURRENT_MENU_ID, currentMenuId)
    }

    private fun switchToSelectedTab(selectedMenuId: Int) {
        var isCreated = false
        var selectedTabId = 0
        when (selectedMenuId) {
            R.id.nav_home -> {
                if (homeTabContainer.findViewById<FrameLayout>(R.id.TabContainer) != null) {
                    isCreated = true
                }
                selectedTabId = R.id.homeTabContainer
            }
            R.id.nav_notice -> {
                if (noticeTabContainer.findViewById<FrameLayout>(R.id.TabContainer) != null) {
                    isCreated = true
                }
                selectedTabId = R.id.noticeTabContainer
            }
            R.id.nav_point -> {
                if (pointTabContainer.findViewById<FrameLayout>(R.id.TabContainer) != null) {
                    isCreated = true
                }
                selectedTabId = R.id.pointTabContainer
            }
            R.id.nav_history -> {
                if (historyTabContainer.findViewById<FrameLayout>(R.id.TabContainer) != null) {
                    isCreated = true
                }
                selectedTabId = R.id.historyTabContainer
            }
            R.id.nav_others -> {
                if (othersTabContainer.findViewById<FrameLayout>(R.id.TabContainer) != null) {
                    isCreated = true
                }
                selectedTabId = R.id.othersTabContainer
            }
        }

        if (currentTabId == selectedTabId) {

        } else {
            if (isCreated == false) {
                val selectedTabFragment = TabContainerFragment()
                val bundle = Bundle()
                bundle.putInt(TabContainerFragment.ARG_TAB_ID, selectedTabId)
                selectedTabFragment.setArguments(bundle)

                supportFragmentManager.beginTransaction()
                        .replace(selectedTabId, selectedTabFragment)
                        .commit()
            }

            findViewById<FrameLayout>(selectedTabId).setVisibility(View.VISIBLE)
            var currentTab = findViewById<FrameLayout>(currentTabId)
            if (currentTab != null) { currentTab.setVisibility(View.GONE) }

            currentTabId = selectedTabId
            currentMenuId = selectedMenuId
        }
    }

    /**
     * Switch Fragments only.
     * @param uri Uri
     * @author Shun Komuro
     */
    override fun onFragmentInteraction(uri: Uri) {}
}
